pkg/applicationserver/applicationserver.go

```diff
@@ -643,6 +673,24 @@ func (as *ApplicationServer) attemptDownlinkQueueOp(ctx context.Context, dev *tt
 	}
 }
 
+// initializeConfirmationRetriesConfig initializes the confirmation retries
+// configuration for the given downlink items.
+func (as *ApplicationServer) initializeConfirmationRetriesConfig(items []*ttnpb.ApplicationDownlink) {
+	maxRetries := as.config.Downlinks.ConfirmationConfig.DefaultRetryAttempts
+	if as.config.Downlinks.ConfirmationConfig.MaxRetryAttempts != 0 {
```

Comment:
This is wrong, because you always use max instead of default when max is not zero.